lib/matplotlib/sphinxext/plot_directive.py

```diff
@@ -519,7 +548,7 @@ def get_plot_formats(config):
 
 def render_figures(code, code_path, output_dir, output_base, context,
                    function_name, config, context_reset=False,
-                   close_figs=False):
+                   close_figs=False, outname=''):
```

Comment:
Would go with `outname=None`, which is IMHO a more standard way for "not defined" than an empty string.